src/main.cpp

```
@@ -155,14 +155,23 @@ int main(int argc, char* argv[]) {
 
   // Auto-detect model for sync mode (skip for mock provider)
   if (cfg.model == "auto" && cfg.mode == Mode::Sync && cfg.provider != "mock") {
-    auto models = provider->list_models();
-    if (!models.empty()) {
-      cfg.model = models[0];
-      Config::instance().model = models[0];
+    // Prefer already-loaded model to avoid cold-start delay (ADR-112)
+    std::string running = get_running_model(cfg);
+    if (!running.empty()) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)